core/simulator.cpp

```diff
@@ -78,12 +79,34 @@ int Simulator::numObjects() const {
   return system->numObjects();
 }
 
-int Simulator::numMovements() const {
+QVariant Simulator::metrics() const {
   QMutexLocker locker(&system->mutex);
-  return system->numMovements();
+  QList<QVariant> metricsData;
+  for (const auto& c : system->getCounts()) {
+    metricsData.push_back(QVariant({c->_name, c->_value}));
+  }
+  for (const auto& m : system->getMeasures()) {
+    if (m->_history.empty()) {
+      metricsData.push_back(QVariant({m->_name, 0.0}));
+    } else {
+      metricsData.push_back(QVariant({m->_name, m->_history.back()}));
+    }
+  }
+  return QVariant::fromValue(metricsData);
 }
 
-int Simulator::numRounds() const {
+void Simulator::exportMetrics() {
   QMutexLocker locker(&system->mutex);
-  return system->numRounds();
+  QFile outFile(QDir::currentPath() + "/metrics_" +
+                QString::number(QDateTime::currentSecsSinceEpoch()) + ".json");
+  if (!outFile.open(QIODevice::WriteOnly | QIODevice::Text)) {
+    return;
+  }
+  QTextStream outStream(&outFile);
+  outStream << system->metricsAsJSON();
+}
```

Comment:
@josephbriones No need to update the copyrights in this PR, though I should definitely do that soon. As for the JSON generation, not sure what would be going wrong. This is the relevant code; as you can see in lines 100-101, everything is being done in platform-independent Qt libraries. According to the documentation, `"/"` is always translated to the correct directory separator if a `QFile` is used.

Reply:
@josephbriones What happens when you try to generate the file? Does it crash? Or does it just not appear where you expect it to?

Reply:
After talking to Joseph, we found that a JSON file is indeed being written. However, `QDir::currentPath()` on macOS is pointing at `<builddir>/AmoebotSim.app/Contents/MacOS`, which is a hidden directory that isn't directly accessible by Finder. I'm investigating how we could patch this to point at a stable location on Windows and macOS so that users will know where to go look.

Reply:
Ok, I think I've found a solution I'm happy with. @josephbriones try replacing `QDir::currentPath()` with `QCoreApplication::applicationDirPath()`. This should be the `QString` path for the directory containing the executable, which is stable and visible on both Windows and macOS. If this works for you, I'll go ahead and push a new commit.